odoo/modules/registry.py

```diff
@@ -363,6 +362,22 @@ def clear_caches(self):
         for model in self.models.values():
             model.clear_caches()
 
+    # The flags 'registry_invalidated' and 'cache_invalidated' are thread-local.
+    # This avoids situations like: thread A modifies the registry, thread B
+    # fails (because of concurrent changes) and resets the registry, while
+    # thread A terminates without signaling its changes...
+    registry_invalidated = property(
+        lambda self: getattr(self._invalidate_flags, 'registry', False),
+        lambda self, value: setattr(self._invalidate_flags, 'registry', value),
+        doc="Determine whether the current thread has modified the registry.",
+    )
+
+    cache_invalidated = property(
+        lambda self: getattr(self._invalidate_flags, 'cache', False),
+        lambda self, value: setattr(self._invalidate_flags, 'cache', value),
+        doc="Determine whether the current thread has modified the cache.",
+    )
```

Comment:
Seems unnecessary to avoid the decorator form. I also don't think the comment is necessary  as (long as) the reason / issue / purpose is explained in the commit message. The few readers which would come around here will hopefully realise there's a reason for those being threadlocals.